platform-tests/src/test/java/org/junit/platform/testkit/engine/EventsTests.java

```diff
@@ -0,0 +1,285 @@
+		events.assertEventsMatchExactly( //
+			event(engine(), started()), //
+			event(engine(), skippedWithReason("reason1")), //
+			event(engine(), skippedWithReason("reason2")), //
+			event(engine(), finishedSuccessfully()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: all events in order -> match")
+	void assertEventsMatchLooselyMatchesAllEventsInOrder() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), started()), //
+			event(engine(), skippedWithReason("reason1")), //
+			event(engine(), skippedWithReason("reason2")), //
+			event(engine(), finishedSuccessfully()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: all events in wrong order -> match")
+	void assertEventsMatchLooselyMatchesAllEventsInWrongOrder() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), skippedWithReason("reason2")), //
+			event(engine(), finishedSuccessfully()), //
+			event(engine(), skippedWithReason("reason1")), //
+			event(engine(), started()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: tailing subset -> match")
+	void assertEventsMatchLooselyMatchesATailingSubset() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), skippedWithReason("reason1")), //
+			event(engine(), finishedSuccessfully()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: starting subset -> match")
+	void assertEventsMatchLooselyMatchesAStartingSubset() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), started()), //
+			event(engine(), skippedWithReason("reason1")) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: subset in wrong order -> match")
+	void assertEventsMatchLooselyMatchesASubsetInWrongOrder() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), skippedWithReason("reason1")), //
+			event(engine(), started()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: only last event -> match")
+	void assertEventsMatchLooselyMatchesTheLastEventAlone() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), finishedSuccessfully()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: only first event -> match")
+	void assertEventsMatchLooselyMatchesTheFirstEventAlone() {
+		events.assertEventsMatchLoosely( //
+			event(engine(), started()) //
+		);
+	}
+
+	@Test
+	@DisplayName("assertEventsMatchLoosely: only bad events -> fails")
+	void assertEventsMatchLooselyWithBadConditionsOnlyFails() {
+		Executable willFail = () -> events.assertEventsMatchLoosely( //
+			event(engine(), finishedWithFailure()), //
+			event(engine(), skippedWithReason("other")) //
+		);
+
+		AssertJMultipleFailuresError error = assertThrows(AssertJMultipleFailuresError.class, willFail);
+
+		List<Throwable> failures = error.getFailures();
+		assertEquals(2, failures.size());
+		assertEquals(AssertionError.class, failures.get(0).getClass());
+		assertEquals(AssertionError.class, failures.get(1).getClass());
```

Comment:
Given that this would indeed be shorter but not really enhance readability I tend to leave it the way it is. The combination of varargs (array) covariance with Class<>-generics also summons trouble...

Reply:
Whether it enhances readability is of course debatable.

But... you typically get more informative error messages if such a construct fails when using AssertJ like that.

Anyway, it works _as-is_. ;-)